### PR TITLE
add metadata to the vspheremachinetemplate

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What this PR does / why we need it**: This is a PR that fixes a breaking change for anyone using annotations in the `VSphereMachineTemplate` introduced in `v0.7.10`. `v0.8.0` didn't have this PR which means that upgrading dropped the `ObjectMeta` field

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: api/v1alpha3/vspheremachinetemplate_conversion.go [hunk 1]
```diff
@@ -17,7 +17,10 @@ limitations under the License.
 package v1alpha3
 
 import (
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	infrav1alpha4 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha4"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	clusterv1a4 "sigs.k8s.io/cluster-api/api/v1alpha4"
```
Comment (srm09) [inline api/v1alpha3/vspheremachinetemplate_conversion.go:23]:
```suggestion
	clusterv1a3 "sigs.k8s.io/cluster-api/api/v1alpha3"
	clusterv1a4 "sigs.k8s.io/cluster-api/api/v1alpha4"
```
nit: This is one of the few places we are using both the imports, makes sense to keep it clear.

File: api/v1alpha3/conversion_test.go [hunk 2]
```diff
@@ -40,17 +41,18 @@ func TestFuzzyConversion(t *testing.T) {
...
 		Spoke:  &VSphereMachine{},
 	}))
 	t.Run("for VSphereMachineTemplate", utilconversion.FuzzTestFunc(utilconversion.FuzzTestFuncInput{
-		Scheme: scheme,
-		Hub:    &nextver.VSphereMachineTemplate{},
-		Spoke:  &VSphereMachineTemplate{},
+		Scheme:      scheme,
+		Hub:         &nextver.VSphereMachineTemplate{},
+		Spoke:       &VSphereMachineTemplate{},
+		FuzzerFuncs: []fuzzer.FuzzerFuncs{CustomObjectMetaFuzzFunc},
```
Comment (srm09) [inline api/v1alpha3/conversion_test.go:55]: nit: Can we use the `CustomObjectMetaFuzzFunc` function from CAPI directly?

**Review by srm09 — COMMENTED** [review]
Minor nits, otherwise lgtm

File: api/v1alpha3/conversion_test.go (hunk 2 above)
Comment (yastij) [reply api/v1alpha3/conversion_test.go:55]: it's not importable, I'll open an issue to move it to some util

**Review by gab-satchi — APPROVED** [review]

**Pull request merged**